tempesta_fw/ss_skb.c

```diff
@@ -313,8 +368,17 @@ __split_linear_data(struct sk_buff *skb, char *pspt, int len, TfwStr *it)
 	 */
 	if (unlikely((len < 0) && (tail_len == -len))) {
 		ss_skb_put(skb, len);
-		if (skb_is_nonlinear(skb))
+		if (skb_is_nonlinear(skb)) {
 			it->ptr = skb_frag_address(&skb_shinfo(skb)->frags[0]);
+			it->skb = pskb ? pskb : skb;
+		} else {
+			struct sk_buff *dskb = __next_skb_frag(skb, pskb);
+			if (dskb)
+				it->skb = pskb ? pskb : skb;
```

Comment:
Intention about ```it->skb``` is simple: when we process ```it->ptr``` we could wish to change/examine corresponding ```skb```, so ```it->skb``` points to appropriate ```skb```. Sure parent skb also contain the data pointed by ```it->ptr```, but we'll lose CPU cycles to find the ```skb``` fragment. So ```it->ptr``` should be either in linear or paged data, not in fragments.